www/modules/custom/borg_project_metrics/borg_project_metrics.module

```diff
@@ -84,4 +90,284 @@ function borg_project_metrics_cron() {
+      try {
+        $last_date = _get_display_metric_from_db('comments')['date'];
+        _parse_metrics('comments', $last_date);
+      }
+      catch(Exception $e) {
+        $message = $e->getMessage();
+        watchdog('borg_project_metrics', $message, array(), WATCHDOG_ERROR);
+      }
+    }
+  }
+}
+
+/**
+ * Get the number of commits to Backdrop since 15 January 2015.
+ *
+ * @param string $type
+ * String to indicate what metric to get: {users, commits, comments}.
+ *
+ * @param string $date_time
+ * The date from DB borg_project_metrics table to set as since for the get.
+ *
+ * @return array $metrics
+ * Returns an array of json objects each element is a paginated response from the
+ * github api.
+ *
+ * @see _get_contrib_comments()
+ */
+function _get_metrics(string $type, string $date_time = '2017-06-01T00:00:00Z') {
+  global $settings;
+  $token = $settings['githubapi_token'];
+  $authorization = "Authorization: token $token";
+  $metrics = [];
+  $i = 1;
+  $date_time = str_replace(' ', 'T', $date_time) . 'Z';
+  do {
+    if ($type == 'users') {
+      $url = "https://api.github.com/repos/backdrop/backdrop/commits?since=$date_time&page=$i";
+    }
+    elseif ($type == 'commits') {
+      $url = "https://api.github.com/repos/backdrop/backdrop/commits?since=$date_time&page=$i";
+    }
+    elseif ($type = 'comments') {
+      $url = "https://api.github.com/repos/backdrop/backdrop-issues/issues/comments?since=$date_time&page=$i";
+    }
+
+    // Initiate a curl session.
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json', $authorization));
+    curl_setopt ($ch, CURLOPT_USERAGENT, "Mozilla/4.0 (compatible; MSIE 5.01; Windows NT 5.0)");
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 1);
+    $metrics[$i] = curl_exec($ch);
+    $res_code = curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
+    curl_close($ch);
+    $json = json_decode($metrics[$i]);
+    $i++;
+  } while ($res_code == 200 && !empty($json));
+
+  // If the $type is comments then we also want to get the contrib comments.
+  if ($type == 'comments') {
+    $project_nodes = _get_project_nodes();
+    foreach ($project_nodes as $m) {
+      $project = explode('/', $m[0]);
+      $project = $project[1];
+      if ($project != 'backdrop' && !empty($project)) {
+        $project_comment_metrics = _get_contrib_comments($project, $date_time);
+        $metrics = array_merge($metrics, $project_comment_metrics);
+      }
+    }
+  }
+
+  return $metrics;
+}
+
+/**
+ * Helper function to parse data from githhub api _get_metrics() request and
+ * store the parsed data in backdrop_project_metrics DB table.
+ *
+ * @param string $type
+ * The type {users, commits, comments} of metric to parse.
+ *
+ * @see _get_display_metric_from_db()
+ * @see _get_metrics()
+ * @see _put_metrics_into_db()
+ */
+function _parse_metrics(string $type) {
```

Comment:
we should properly namespace these helper functions. `borg_project_metrics_parse_metrics()`